src/compiler/checker.ts

```diff
@@ -24001,21 +24021,28 @@ namespace ts {
             if (isNodeOpeningLikeElement) {
                 checkGrammarJsxElement(<JsxOpeningLikeElement>node);
             }
+
             checkJsxPreconditions(node);
             // The reactNamespace/jsxFactory's root symbol should be marked as 'used' so we don't incorrectly elide its import.
             // And if there is no reactNamespace/jsxFactory's symbol in scope when targeting React emit, we should issue an error.
-            const reactRefErr = diagnostics && compilerOptions.jsx === JsxEmit.React ? Diagnostics.Cannot_find_name_0 : undefined;
-            const reactNamespace = getJsxNamespace(node);
-            const reactLocation = isNodeOpeningLikeElement ? (<JsxOpeningLikeElement>node).tagName : node;
-            const reactSym = resolveName(reactLocation, reactNamespace, SymbolFlags.Value, reactRefErr, reactNamespace, /*isUse*/ true);
-            if (reactSym) {
+            const jsxFactoryRefErr = diagnostics && compilerOptions.jsx === JsxEmit.React ? Diagnostics.Cannot_find_name_0 : undefined;
+            const jsxFactoryNamespace = getJsxNamespace(node);
+            const jsxFactoryLocation = isNodeOpeningLikeElement ? (<JsxOpeningLikeElement>node).tagName : node;
+
+            // allow null as jsxFragmentFactory

```

Comment:
Just curious, what’s the utility of this? I couldn’t find anything about using `null` in the babel transform docs.

Reply:
I think the idea is that some libraries may use a `null` tag name to indicate a fragment, rather than some specific fragment object. [Mithril actually uses `'['` as the fragment nowadays](https://github.com/MithrilJS/mithril.js/blob/720cccf73a049cdabcc77c969a24d30225cc2582/docs/jsx.md), so we should actually consider supporting arbitrary simple expressions here...

Reply:
snabbdom-jsx-lite uses null. https://github.com/nojvek/snabbdom-jsx-lite/blob/master/tsconfig.json#L8

Mostly because if <div /> is compiled to `h('div', null)`. null meaning no attrs,

`<>` compiles to `h(null, null)` meaning it's no named tag i.e Fragment. This avoids an extra function call for evaluation of fragment function, which in most cases convert the children to an array.